src/ffi/client.rs

```diff
@@ -95,6 +95,7 @@ ffi_fn! {
     fn hyper_clientconn_options_new() -> *mut hyper_clientconn_options {
         let mut builder = conn::Builder::new();
         builder.http1_preserve_header_case(true);
+        builder.http1_preserve_header_order(true);
```

Comment:
Should I remove this line (making an ffi connection not preserve ordering by default)?

Reply:
I'm thinking we'll eventually want to make both of these options default off. Just because curl does want these doesn't mean all users of the C API want to pay the cost...

Reply:
No that I've added the setters for both of these, I'll just yank it out now. I'll put up a PR to curl once this has merged.